ui/app/webpack.common.ts

```diff
@@ -12,76 +12,40 @@
 // limitations under the License.
 
 import path from 'path';
-import { Configuration } from 'webpack';
 import { CleanWebpackPlugin } from 'clean-webpack-plugin';
 import HtmlWebpackPlugin from 'html-webpack-plugin';
-import ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
-
-// File paths used by the build
-const ROOT = path.resolve(__dirname);
-export const FilePaths = {
-  ROOT,
-  SRC: path.resolve(ROOT, 'src'),
-  OUT: path.resolve(ROOT, 'dist'),
-};
+import { Configuration } from 'webpack';
 
-// Webpack configuration that's common between environments
 export const commonConfig: Configuration = {
-  context: FilePaths.SRC,
-  entry: {
-    main: './index.tsx',
-  },
-
+  entry: path.resolve(__dirname, './src/index.tsx'),
   output: {
-    path: FilePaths.OUT,
+    path: path.resolve(__dirname, './dist'),
     publicPath: '/',
   },
-
   resolve: {
-    // Add ts/tsx to list of files that can be imported without including the extension
-    extensions: ['.ts', '.tsx', '.js', '.jsx', '.json'],
+    extensions: ['.ts', '.tsx', '.js', 'jsx', '.json'],
   },
-
   plugins: [
-    new CleanWebpackPlugin(),
+    new CleanWebpackPlugin({
+      dangerouslyAllowCleanPatternsOutsideProject: true,
+    }),
 
     // Generates HTML index page with bundle injected
     new HtmlWebpackPlugin({
-      template: path.resolve(FilePaths.SRC, 'index.html'),
+      template: path.resolve(__dirname, './src/index.html'),
       templateParameters: {},
     }),
-
-    // Does TS type-checking in a separate process
-    new ForkTsCheckerWebpackPlugin({
-      typescript: {
-        configFile: path.resolve(FilePaths.ROOT, 'tsconfig.json'),
-      },
-    }),
   ],
-
-  // How to load and transpile modules
   module: {
     rules: [
       {
         test: /\.tsx?$/,
-        include: [FilePaths.SRC],
-        enforce: 'pre',
-        use: [
-          {
-            loader: 'eslint-loader',
-          },
-        ],
-      },
-
-      {
-        test: /\.tsx?$/,
-        //include: [FilePaths.SRC],
         use: [
           {
             loader: 'ts-loader',
             options: {
               // Type-checking happens in separate plugin process
-              transpileOnly: true,
```

Comment:
Doing type-checking in a separate process/plugin is supposed to make the build faster. Is that not necessary any more?